spec/java/security/SecureRandom.main.lsl

```diff
@@ -0,0 +1,885 @@
+import java/util/stream/IntStream;
+import java/util/stream/LongStream;
+
+
+// automata
+
+automaton SecureRandomAutomaton
+(
+    var provider: Provider,
+    var algorithm: String,
+    var defaultProvider: boolean
+)
+: SecureRandom
+{
+    // states and shifts
+
+    initstate Allocated;
+    state Initialized;
+
+    shift Allocated -> Initialized by [
+        // constructors
+        SecureRandom (SecureRandom),
+        SecureRandom (SecureRandom, SecureRandomSpi, Provider),
+        SecureRandom (SecureRandom, SecureRandomSpi, Provider, String),
+        SecureRandom (SecureRandom, array<byte>),
+        // static operations
+        getInstance (String),
+        getInstance (String, Provider),
+        getInstance (String, String),
+        getInstanceStrong,
+        getSeed,
+    ];
+
+    shift Initialized -> self by [
+        // instance methods
+        doubles (SecureRandom),
+        doubles (SecureRandom, double, double),
+        doubles (SecureRandom, long),
+        doubles (SecureRandom, long, double, double),
+        generateSeed,
+        getAlgorithm,
+        getProvider,
+        ints (SecureRandom),
+        ints (SecureRandom, int, int),
+        ints (SecureRandom, long),
+        ints (SecureRandom, long, int, int),
+        longs (SecureRandom),
+        longs (SecureRandom, long),
+        longs (SecureRandom, long, long),
+        longs (SecureRandom, long, long, long),
+        nextBoolean,
+        nextBytes (SecureRandom, array<byte>),
+        nextDouble,
+        nextFloat,
+        nextGaussian,
+        nextInt (SecureRandom),
+        nextInt (SecureRandom, int),
+        nextLong,
+        setSeed (SecureRandom, array<byte>),
+        setSeed (SecureRandom, long),
+    ];
+
+
+    // utilities
+
+    @AutoInline @Phantom proc _throwNPE (): void
+    {
+        action THROW_NEW("java.lang.NullPointerException", []);
+    }
+
+
+    @AutoInline @Phantom proc _throwIAE (): void
+    {
+        action THROW_NEW("java.lang.IllegalArgumentException", []);
+    }
+
+
+    @AutoInline @Phantom proc _throwNSAE (): void
+    {
+        action THROW_NEW("java.security.NoSuchAlgorithmException", []);
+    }
+
+
+    @AutoInline @Phantom proc _throwIE (): void
+    {
+        action THROW_NEW("java.lang.InternalError", []);
+    }
+
+
+    @AutoInline @Phantom proc _throwIAE (): void
+    {
+        action THROW_NEW("java.lang.IllegalArgumentException", []);
+    }
+
+
+    @AutoInline @Phantom proc _throwNSPE (): void
+    {
+        action THROW_NEW("java.security.NoSuchProviderException", []);
+    }
+
+
+    proc _getDefaultPRNG (setSeed: boolean, seed: array<byte>): void
+    {
+        val providersList: array<Provider> = action CALL_METHOD(null as Security, "getProviders", []);
+        val providersListLength: int = action ARRAY_SIZE(providersList);
+
+        var i: int = 0;
+        action LOOP_FOR(
+            i, 0, providersListLength, +1,
+            findProvider_loop(i, providersList)
```

Comment:
@dog-m I think it must have such suffix: "loop"; I have fixed in such way: "_getDefaultPRNG_findProvider_loop"

Reply:
Loop was deleted, because were used:

```
action SYMBOLIC("java.security.Provider");
action SYMBOLIC("java.lang.String");
```

temporary decision;